### PR TITLE
Feature/completed task

### PR DESCRIPTION
**Pull request by CibelePaulinoAndrade**
partial done : completed task
refactor neon frame view (to suport alerts)

### CODE REVIEW

File: FinalChallenge/AppDelegate.swift [hunk 1]
```diff
@@ -16,10 +16,20 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+        let progress: String = "00"
+        let message: String = "Yay! Task completed in time!"
+        
+        let viewModel: CompletedTaskViewModel = CompletedTaskViewModel.init(message: message, progress: PrizeDisplayType.label(progress: progress), furniture: PrizeDisplayType.image(furnitureImage: image))
+        
         window = UIWindow(frame: UIScreen.main.bounds)
-        window?.rootViewController = ViewController()
+
+        window?.rootViewController = CompletedTaskViewController(viewModel: viewModel)
+//        window?.rootViewController = ViewController()
```
Comment (jessicalewinter) [inline FinalChallenge/AppDelegate.swift]: remove comment or remove your viewController from app delegate

File: FinalChallenge/Utils/Protocol/UIBezierPathable.swift [hunk 2]
```diff
@@ -9,6 +9,7 @@
 import UIKit
 
 protocol UIBezierPathable {
+
```
Comment (jessicalewinter) [inline FinalChallenge/Utils/Protocol/UIBezierPathable.swift]: remove space here

File: FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift [hunk 3]
```diff
@@ -9,11 +9,25 @@
...
-    private var viewTitle: String
-    lazy var titleLabel = UILabel.init(textTitle: viewTitle)
+    private var viewTitle: String?
+    private var withCloseButton: Bool
+    var withTitle: Bool
+    
+    var layoutGuide: UILayoutGuide = UILayoutGuide()
+    
+    lazy var titleLabel: UILabel = {
+        guard let text = viewTitle else { fatalError("Nil value was found!") }
```
Comment (jessicalewinter) [inline FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift]: Does it need to have a fatal error? Perhaps just a empty string. Fatal error is not always good.

File: FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift [hunk 4]
```diff
@@ -57,14 +78,30 @@ class NeonFrameView: UIView {
     /// - Parameters:
     ///   - titleView: Name of view to be performed
     ///   - containerView: View that will compose the main structure
-    init(titleView: String, containerView: UIView) {
+    init(titleView: String?, containerView: UIView, withCloseButton: Bool = true, withTitle: Bool = true) {
```
Comment (jessicalewinter) [inline FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift]: titleView and viewTitle? I think you should use the name viewTitle to make it more consistent

File: FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift [hunk 5]
```diff
@@ -57,14 +78,30 @@ class NeonFrameView: UIView {
...
         setupView()
     }
     
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
+    
+    override func didMoveToSuperview() {
+        guard let parent = superview else {
+            fatalError("Superview was nil")
```
Comment (jessicalewinter) [inline FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift]: Do you need to throw a fatal error? Does you always need to have a superview?

File: FinalChallenge/Views/Completed Task View/CompletedTaskView.swift [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    /// Function called when save button is clicked
+    @objc func saveButtonIsTapped(_ sender: UITapGestureRecognizer? = nil) {
+        delegate?.completedTaskViewDoneButtonHasBeenTouched()
+    }
+    
+    func animateView() {
+        self.handImageView.alpha = 0.2
+        UIView.animate(withDuration: 0.5, delay: 0.0, options: [.curveLinear, .repeat, .autoreverse], animations: {
+            self.handImageView.alpha = 1.0
+
```
Comment (jessicalewinter) [inline FinalChallenge/Views/Completed Task View/CompletedTaskView.swift]: remove space

File: FinalChallenge/Views/NeonView/Extension/UINeonView+TopAcessoryWithoutCloseButton.swift [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+//
+//  Created by Cibele Paulino on 25/10/19.
+//  Copyright © 2019 Elias Paulino. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+extension UINeonView {
+
```
Comment (jessicalewinter) [inline FinalChallenge/Views/NeonView/Extension/UINeonView+TopAcessoryWithoutCloseButton.swift]: remove space

File: FinalChallenge/Views/Prize Display/NeonRoundRect.swift [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+//
+//  Created by Cibele Paulino on 27/10/19.
+//  Copyright © 2019 Elias Paulino. All rights reserved.
+//
+
+import Foundation
+import UIKit
+/// Create a rounded neon view
+class NeonRoundRect: UIView, Neonable {
+
```
Comment (jessicalewinter) [inline FinalChallenge/Views/Prize Display/NeonRoundRect.swift]: remove space

File: FinalChallenge/Views/Prize Display/NeonRoundRect.swift [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+class NeonRoundRect: UIView, Neonable {
+    
+    /// define the path of a neon line
+    var neonPath: CGPath {
+        return UIBezierPath(roundedRect: self.neonSafeBounds, byRoundingCorners: .allCorners, cornerRadii: CGSize.init(width: 10, height: 10)).cgPath
+    }
+    
+    /// Defines the neon color
+    var neonColor: UIColor = UIColor.action
+
```
Comment (jessicalewinter) [inline FinalChallenge/Views/Prize Display/NeonRoundRect.swift]: remove space between variables

File: FinalChallenge/Views/Prize Display/NeonRoundRect.swift [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    var neonColor: UIColor = UIColor.action
+    
+    var pathFillColor: UIColor = UIColor.action
+    
+    ///Defines the center of the view.
+    var centerXConstraint: NSLayoutConstraint?
+    var centerYConstraint: NSLayoutConstraint?
+    
+    init(borderColor: UIColor = UIColor.action) {
+        self.neonColor = borderColor
```
Comment (jessicalewinter) [inline FinalChallenge/Views/Prize Display/NeonRoundRect.swift]: you should rename neonColor or borderColor to understand what you're trying to do

File: FinalChallenge/AppDelegate.swift (hunk 1 above)
Comment (CibelePaulinoAndrade) [reply FinalChallenge/AppDelegate.swift]: done

File: FinalChallenge/Utils/Protocol/UIBezierPathable.swift (hunk 2 above)
Comment (CibelePaulinoAndrade) [reply FinalChallenge/Utils/Protocol/UIBezierPathable.swift]: done

File: FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift (hunk 3 above)
Comment (CibelePaulinoAndrade) [reply FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift]: Within the way I designed the view, it is possible not to get a value for the title even when it is said that you want this view with the title, so I put the fatalError indicating what is wrong if the person tries to use something that is not. It's the right thing.

File: FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift (hunk 4 above)
Comment (CibelePaulinoAndrade) [reply FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift]: done

File: FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift (hunk 5 above)
Comment (CibelePaulinoAndrade) [reply FinalChallenge/Views/ Neon Frame View/NeonFrameView.swift]: done

File: FinalChallenge/Views/Completed Task View/CompletedTaskView.swift (hunk 6 above)
Comment (CibelePaulinoAndrade) [reply FinalChallenge/Views/Completed Task View/CompletedTaskView.swift]: done

File: FinalChallenge/Views/NeonView/Extension/UINeonView+TopAcessoryWithoutCloseButton.swift (hunk 7 above)
Comment (CibelePaulinoAndrade) [reply FinalChallenge/Views/NeonView/Extension/UINeonView+TopAcessoryWithoutCloseButton.swift]: done

File: FinalChallenge/Views/Prize Display/NeonRoundRect.swift (hunk 8 above)
Comment (CibelePaulinoAndrade) [reply FinalChallenge/Views/Prize Display/NeonRoundRect.swift]: done

File: FinalChallenge/Views/Prize Display/NeonRoundRect.swift (hunk 9 above)
Comment (CibelePaulinoAndrade) [reply FinalChallenge/Views/Prize Display/NeonRoundRect.swift]: done

File: FinalChallenge/Views/Prize Display/NeonRoundRect.swift (hunk 10 above)
Comment (CibelePaulinoAndrade) [reply FinalChallenge/Views/Prize Display/NeonRoundRect.swift]: I used this name in init for anyone who will use to understand where the color will be used, which is on the border. It is not possible to rename neonColor as it is from a protocol.

Comment (jessicalewinter) [reply FinalChallenge/Views/Prize Display/NeonRoundRect.swift]: Ok, then

**Pull request merged**